contracts/spend-limit/src/authenticator/common.rs

```diff
@@ -0,0 +1,59 @@
+use cosmwasm_std::{Addr, Coin, DepsMut, Timestamp, Uint128};
+
+use crate::{
+    period::Period,
+    price::{get_and_cache_price, PriceResolutionConfig},
+    spend_limit::Spending,
+    state::PRICE_INFOS,
+    ContractError,
+};
+
+pub fn get_account_spending_fee(
+    account: &Addr,
+    fee_payer: &Addr,
+    fee_granter: Option<&Addr>,
+    fee: Vec<Coin>,
+) -> Vec<Coin> {
+    if let Some(fee_granter) = fee_granter {
+        if account == fee_granter {
+            fee
+        } else {
+            vec![]
+        }
+    } else {
+        if account == fee_payer {
+            fee
+        } else {
+            vec![]
+        }
+    }
```

Comment:
I'd like to see this written differently, but that's just a stylistic choice

Reply:
yeah maybe refactor them into composing boolean exp and single if branch would be better. Will do that.